docs/fastdds/ros2/Discovery_service/ros2_discovery_service.rst

```diff
@@ -0,0 +1,337 @@
+
+The following tutorial gathers the steps to check this functionality and learn how to use it with ROS2.
+
+.. contents::
+    :local:
+    :backlinks: none
+    :depth: 2
+
+The :ref:`Simple Discovery Protocol <simple_disc_settings>` is the
+standard protocol defined in the `standard DDS <https://www.omg.org/omg-dds-portal/>`__.
+But it has certain known disadvantages in some scenarios:
+
+* It does not **Scale** efficiently, as the number of exchanged packets highly increases as new nodes are added.
+* It requires **Multicasting** capabilities, that may not work reliably in some connection cases as WiFi.
+
+The **Discovery Service** provides a Client-Server Architecture that allows
+the nodes to connect with each other using an intermediate server.
+Each node will work as a *Client*, sharing its info with the *Discovery Server* and receiving
+the discovery information from it.
+This means that the network traffic is highly reduced in big systems, and it does not require *Multicasting*.
+
+.. image:: ds_explanation.png
+    :align: center
+
+These **Discovery Servers** can be independent, duplicated or connected with each other in order to create
+redundancy over the network and avoid having a *Single-Point-Of-Failure*.
+
+In order to use this functionality, **Fast-DDS Discovery Service** can be set by parameters using
+the :ref:`XML configuration for Participants <DS_setup_concepts>`.
+
+However, Fast DDS library provides an easier way to set a **Discovery Service** communication using
+the ``fastdds`` :ref:`CLI tool <ffastddscli_cli>`
+and an :ref:`environment variable <env_vars>`.
+
+For a complete information about the actual implementation, testing and performance,
+visit this `link <https://eprosima.com/index.php/resources-all/scalability/fast-rtps-discovery-mechanisms-analysis>`__
+
+
+Prerequisites
+-------------
+
+This tutorial assumes you have a `working ROS 2 installation <https://index.ros.org/doc/ros2/Installation/>`__.
+In case your installation is using a Fast DDS version lower than v2.1.0, you could not use the ``fastdds`` tool.
+You could update your repository to use a different Fast DDS version,
+or :ref:`set the discovery service by Fast-DDS XML QoS configuration <discovery_service>`.
+
+The ``talker-listener`` ROS2 demo allows to create a *talker* node that publishes a *Hello World* message every second,
+and a *listener* node that listens to these messages.
+
+
+Run the demo
+------------
+
+`Sourcing ROS 2 <https://index.ros.org/doc/ros2/Tutorials/Configuring-ROS2-Environment/>`__
+you will get access to the CLI of Fast DDS library CLI tool ``fastdds``.
+From this CLI we have access to the :ref:`discovery tool <cli_discovery>`,
+which allows to launch a server. This server will manage the discovery process for the nodes that connect to it.
+
+.. note::
+
+    Do not forget to `source ROS 2 <https://index.ros.org/doc/ros2/Tutorials/Configuring-ROS2-Environment/>`__
+    in every new terminal you open.
```

Comment:
Better use reported speech

Reply:
This again comes from the use of speech in ROS 2 Index Tutorials. However, I do not see this speech use as a problem.